pkg/client/cache/delta_fifo.go

```diff
@@ -522,6 +522,23 @@ func (f *DeltaFIFO) syncKey(key string) error {
 		return nil
 	}
 
+	// If we are doing Resync() and there is already an event queued for that object,
```

Comment:
I don't understand your comment. Basically we are calling Replace e.g. when we are doing "relist", if we are not able to get all the deltas from in-between. This is effectively an "update", but this is handled in the code as "Sync".